solr/core/src/java/org/apache/solr/parser/SolrQueryParserBase.java

```diff
@@ -539,6 +564,25 @@ protected Query newRegexpQuery(Term regexp) {
     return query;
   }
 
+  @Override
+  protected Query newSynonymQuery(Term terms[]) {
+    if (scoreOverlaps == ScoreOverlaps.PICK_BEST) {
```

Comment:
Some nitpicks here.  I think a switch/case statement would better reflect this code reacts to all possibilities of scoreOverlaps.  Secondly, the `new ArrayList<Query>()` could be `new ArrayList<>(terms.length)`.